### PR TITLE
chore: use targetNamespace for authorino feature

### PR DESCRIPTION
**Pull request by cam-garrison**
## Description

This PR uses TargetNamespace for the authorino feature, removing the need for the helper ResolveAuthNamespace() function. 

## How Has This Been Tested?

Tested on CRC with image `quay.io/cgarriso/opendatahub-operator:dev-use-target-ns`, created DSCI with default auth namespace and saw that authorino pod was created in targetNS `opendatahub-auth-provider` as expected and then specified in DSCI with namespace `auth-provider` to ensure specifying still works. 

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: controllers/dscinitialization/servicemesh_setup.go [hunk 1]
```diff
@@ -94,8 +94,12 @@ func configureServiceMeshFeatures() feature.FeaturesProvider {
 			return cfgMapErr
 		}
 
+		authNs := servicemesh.CreateAuthNamespace(serviceMeshSpec.Auth.Namespace,
+			handler.DSCInitializationSpec.ApplicationsNamespace)
+
 		extAuthzErr := feature.CreateFeature("mesh-control-plane-external-authz").
 			For(handler).
+			TargetNamespace(authNs).
```
Comment (israel-hdez) [inline controllers/dscinitialization/servicemesh_setup.go:102]: I think a change in `mesh-authz-ext-provider.patch.tmpl` is missing:
https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/pkg/feature/templates/servicemesh/authorino/mesh-authz-ext-provider.patch.tmpl#L12

**Review by bartoszmajsak — COMMENTED** [review]
I would suggest putting this PR on hold for the time being, as I am reworking the `f.Spec` stuff entirely.

**Comment by biswassri** [comment]
@cam-garrison Do we still need this PR?

**Comment by cam-garrison** [comment]
No longer needed and underlying code has been reworked.

**Pull request closed without merging**